lib/ruby_llm/utils.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module RubyLLM
+  # Utility methods for the RubyLLM library.
+  module Utils
+    module_function
+
+    def symbolize_keys_deep(hash)
```

Comment:
This is like ActiveSupport's `deep_symbolize_keys`, which I also needed in 2 PRs [here](https://github.com/crmne/ruby_llm/pull/124/files#diff-a7a6f03a46978b8c867bf641262ba489f05e88ef500c160af627077e75496442R59-R76
) and [here](https://github.com/crmne/ruby_llm/pull/131/files#diff-a7a6f03a46978b8c867bf641262ba489f05e88ef500c160af627077e75496442R53-R70)

You'll want to cover Array child items though, so I recommend just taking my/ActiveSupport's implementation and putting it here, then I'll refactor my PRs to use your Util method instead.